### PR TITLE
Install tools update feature product match logic 

### PR DESCRIPTION
**Pull request by jjiwooLim**
Reverts and fixes  https://github.com/OpenLiberty/open-liberty/pull/28549

logs from before:
 ```
% ./featureUtility if batch-2.1          
Initializing ...
Using 8 threads to download artifacts.
Resolving remote features. This process might take several minutes to complete.
CWWKF1300E: The feature batch-2.1 cannot be installed to IBM WebSphere Application Server Liberty Core edition because it only applies to other IBM WebSphere Application Server Liberty editions and Open Liberty. Use the "find" action of the featureManager command to retrieve a list of features that are applicable to IBM WebSphere Application Server Liberty Core edition.

% ./installUtility install batch-2.1
Establishing a connection to the configured repositories ...
This process might take several minutes to complete.

Successfully connected to all configured repositories.

Preparing assets for installation. This process might take several minutes to complete.
CWWKF1256E: The asset batch-2.1 cannot be installed to {1} {2} edition because it only applies to the editions {3}. Use the "find" action of the installUtility command to retrieve a list of assets that are applicable to {4} {5} edition.
```

logs after
```
% ./featureUtility if batch-2.1          
Initializing ...
Resolving remote features. This process might take several minutes to complete.
CWWKF1410E: The batch-2.1 feature cannot be installed to IBM WebSphere Application Server Liberty Liberty Core edition because it only applies to the 

- IBM WebSphere Application Server Liberty Base 
- IBM WebSphere Application Server Liberty - Express 
- IBM WebSphere Application Server Liberty Network Deployment 
- IBM WebSphere Application Server Liberty for Developers 
- IBM Open Liberty 
- IBM WebSphere Application Server Liberty z/OS 

 edition.
% ./installUtility install batch-2.1
Establishing a connection to the configured repositories ...
This process might take several minutes to complete.

Successfully connected to all configured repositories.

Preparing assets for installation. This process might take several minutes to complete.
CWWKF1256E: The asset batch-2.1 cannot be installed to IBM WebSphere Application Server Liberty Liberty Core edition because it only applies to the editions 

- IBM WebSphere Application Server Liberty Base 
- IBM WebSphere Application Server Liberty - Express 
- IBM WebSphere Application Server Liberty Network Deployment 
- IBM WebSphere Application Server Liberty for Developers 
- IBM WebSphere Application Server Liberty z/OS 

. Use the "find" action of the installUtility command to retrieve a list of assets that are applicable to IBM WebSphere Application Server Liberty Liberty Core edition. 
```

### CODE REVIEW

**Comment by jjiwooLim** [comment]
#build
#create.im.repo=true
#create.pkg.images=true

**Comment by jjiwooLim** [comment]
Build not triggered with the properties, kicking off again manually:
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_GnyLYB3EEe-rEslQpyeuMw

**Comment by jjiwooLim** [comment]
#libby-bot

**Comment by jjiwooLim** [comment]
#build
create.im.repo=true
create.pkg.images=true

**Comment by jjiwooLim** [comment]
create.im.repo
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_LQui0CNeEe-4F592xetJGQ

File: dev/wlp.lib.extract/src/wlp/lib/extract/ProductMatch.java [hunk 1]
```diff
@@ -61,13 +63,30 @@ public void add(String substring) {
                     break;
                 }
             }
+        } else if (substring.startsWith("edition")) {
+            //If editions contains Open, add supported WebSphere editions.
+            if (!editions.isEmpty() && editions.contains("Open")) {
+                String editionStr = getValue(substring);
+                for (int startIndex = 0, endIndex = editionStr.indexOf(',');; startIndex = endIndex, endIndex = editionStr.indexOf(',', ++startIndex)) {
```
Comment (jacobwdv) [inline dev/wlp.lib.extract/src/wlp/lib/extract/ProductMatch.java]: We can use Java API's to do this loop instead so it's a little more clear.

```
Collections.addAll(editions,editionStr.split(","));
```

**Review by jacobwdv — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.repository.resolver/src/com/ibm/ws/repository/resolver/RepositoryResolver.java [hunk 2]
```diff
@@ -906,7 +907,21 @@ private void reportErrors() throws RepositoryResolutionException {
...
 
         for (ApplicableToProduct esa : resourcesWrongProduct) {
-            missingRequirements.add(new MissingRequirement(esa.getAppliesTo(), (RepositoryResource) esa));
+            String appliesTo = esa.getAppliesTo();
+            if (appliesTo.contains("productEdition=Open")) { //Check if there are WebSphere editions and add
+                if (esa instanceof EsaResource && ((EsaResource) esa).getAppliesToFilterInfo() != null) {
+                    for (AppliesToFilterInfo atfi : ((EsaResource) esa).getAppliesToFilterInfo()) {
+                        if (!atfi.getEditions().toString().isEmpty()) {
+                            StringBuffer sb = new StringBuffer("; editions=\"");
+                            sb.append(atfi.getEditions().toString().replace("[", "").replace("]", "").replace(" ", ""));
```
Comment (jacobwdv) [inline dev/com.ibm.ws.repository.resolver/src/com/ibm/ws/repository/resolver/RepositoryResolver.java]:
```suggestion
                           sb.append(atfi.getEditions().stream().map(String::strip).collect(Collectors.joining(","));
```

**Review by jacobwdv — APPROVED** [review]
LGTM, Thanks!

**Comment by jjiwooLim** [comment]
#libby-bot

**Pull request merged**